app/server.js

```diff
@@ -0,0 +1,65 @@
+const ttn = require("ttn")
+const app = require("express")()
+const http = require("http").Server(app)
+const bodyParser = require("body-parser")
+
+const googleMapsClient = require("@google/maps").createClient({
+  key: "xxxxxxxxxxxxxxxxxxxxxxxxxx"
+})
+
+var appID = "xxxxxxxxxxxxxxxxx"
+var accessKey = "ttn-account-v2.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+
+var macAddress1 = ""
+var macAddress2 = ""
+var macAddress3 = ""
+
+app.use(bodyParser.json())
+app.use(bodyParser.urlencoded({ extended: false }))
+
+//this is the default home page of our application
+app.get("/", function(req, res){
+  res.sendFile(__dirname + "/map.html")
+})
+
+app.get("/generator", function(req, res){
+  googleMapsClient.geolocate({
+    "considerIp": false,
+    "wifiAccessPoints": [
+      {
+        "macAddress": macAddress1
+      },
+      {
+        "macAddress": macAddress2
+      },
+      {
+        "macAddress": macAddress3
+      }
+    ]
+  }, function(err, response) {
+    console.log(response)
+    console.log(response.json.location)
+    return res.json(response.json.location)
+  })
+})
+
+//opening mqtt connection with ttn
+ttn.data(appID, accessKey)
+.then(function(client) {
+  //on uplink messages
+  client.on("uplink", function(devID, payload) {
+    //print the message in the console
+    console.log("Received uplink from ", devID)
+    console.log(payload.payload_fields)
+    macAddress1 = payload.payload_fields.Bssid1
+    macAddress2 = payload.payload_fields.Bssid2
+    macAddress3 = payload.payload_fields.Bssid3
```

Comment:
What happens if 1 or 2 MAC addresses are received?

Reply:
Also, @Yerion is that a capital `B`? why?

Reply:
Changed capital B to b to fit @Yerion modifications
Also if only 1 or 2 MAC addresses are received, the google API can't give back a proper location, so I'll change the code so that it sends back an error.

Reply:
Is that true yeah?

In any case, you may receive 1 or 2, as the node may see only 1 or 2 APs in the first place

Reply:
I can still try to send requests to the API with only 1 or 2 addresses to make it more open to any case we could encounter